db.go

```diff
@@ -571,3 +572,92 @@ func (db *DB) rewriteValidRecords(walFile *wal.WAL, validRecords []*ValueLogReco
 	}
 	return db.index.PutBatch(positions, matchKeys...)
 }
+
+func (db *DB) NewIterator(options IteratorOptions) (*MergeIterator, error) {
+	if db.options.IndexType != BTree {
+		return nil, ErrDBIteratorUnsupportedType
+	}
+	db.mu.Lock()
+	defer func() {
+		if r := recover(); r != nil {
+			db.mu.Unlock()
+		}
+	}()
+	itrs := make([]*SingleIter, 0, db.options.PartitionNum+len(db.immuMems)+1)
+	rank := 0
+	index := db.index.(*BPTree)
+	for i := 0; i < db.options.PartitionNum; i++ {
+		tx, err := index.trees[i].Begin(false)
+		if err != nil {
+			return nil, err
+		}
+		itr, err := NewBptreeIterator(
+			tx,
+			options,
+		)
+		if err != nil {
+			return nil, err
+		}
+		itr.Rewind()
+		// is empty
+		if !itr.Valid() {
+			itr.Close()
+			continue
+		}
+		itrs = append(itrs, &SingleIter{
+			iType:   BptreeItr,
+			options: options,
+			rank:    rank,
+			idx:     rank,
```

Comment:
There is no same key in disk index, so maybe the rank is not necessary for it?

Reply:
"rank" is used to represent the freshness of data, while "idx" is used to indicate its position in itrs. In heap operations, the index may change, but the rank remains unchanged. Therefore, both are needed.